### ISSUE TITLE
github action for gen_metadata_mini

### ISSUE THREAD

**Issue by thibaulttabarin** [hdr-bgnn/drexel_metadata#3]
We need to create a github action to generate automatically the singularity container defined at https://github.com/hdr-bgnn/drexel_metadata/tree/Thibault/gen_metadata_mini/Singularity_def

**Comment by johnbradley** [comment]
@thibaulttabarin I noticed there is a Pipfile in the root directory of this repo that has different versions of various software than specified in Singularity_def/detectron2_env.def.

For example we install `detectron2==0.4` in singularity:
https://github.com/hdr-bgnn/drexel_metadata/blob/e976898424f789a1e3ffdfcb970ac3c5139e16f7/gen_metadata_mini/Singularity_def/detectron2_env.def#L16

In the Pipfile it installs `detectron2` `v0.6`:
https://github.com/hdr-bgnn/drexel_metadata/blob/e976898424f789a1e3ffdfcb970ac3c5139e16f7/Pipfile#L21

Are the differences intentional or should I try to match the software specified in the Pipfile?

**Comment by thibaulttabarin** [comment]
Yep I know, it was sort of intentional, I try to setup the container many
time with no success and using this version 0.4 was the only configuration
that worked, feel free to try to make version 0.6 work.

**Comment by johnbradley** [comment]
@thibaulttabarin Ok. Using the same version as the training code is best for reproducibility so I'll work towards that.

I also noticed that there are two copies of gen_metadata.py in the Thibault branch:
- [gen_metadata.py](https://github.com/hdr-bgnn/drexel_metadata/blob/Thibault/gen_metadata.py)
- [gen_metadata_mini/scripts/gen_metadata.py](https://github.com/hdr-bgnn/drexel_metadata/blob/Thibault/gen_metadata_mini/scripts/gen_metadata.py)

The container is currently includes the **gen_metadata_mini/scripts/gen_metadata.py** version.
Are you planning on replacing **gen_metadata.py** in the root directory with the contents of **gen_metadata_mini/scripts/gen_metadata.py**?

**Comment by thibaulttabarin** [comment]
ok, sounds for the version of detectron2, the only reason I kept v0.4 was that it didn't work and then I never had to come back to the issue.
For the 2 copies of gen_metadata.py I know it is on my todo list. I haven't decided what to do with this. I think I will rename  gen_metadata_mini/scripts/gen_metadata.py to some think like gen_metadata_mini/scripts/gen_metadata_mini.py.

**Comment by johnbradley** [comment]
I recommend combining them so there are not two copies to maintain. You can see the differences here: https://github.com/hdr-bgnn/drexel_metadata/compare/Thibault...johnbradley:jpb-test?expand=1#diff-6116c0284a7d8756287a91613c46941b05f81d9c8580132da354f7ff9e488704
It looks like your changes are really close to being backwards compatible.
If we need two scripts for some reason you could also abstract out the common parts.

**Comment by johnbradley** [comment]
Fixed by PR #6
